src/formatDate.js

```diff
@@ -8,7 +8,47 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const split = fromFormat[3];
+  const dayOfYear = date.split(split);
+  const tab = [];
+
+  for (let i = 0; i < 3; i++) {
+    for (let j = 0; j < 3; j++) {
+      if (fromFormat[i] === toFormat[j]) {
+        tab[j] = dayOfYear[i];
+      }
+    }
+  }
+
+  let year;
+
+  for (let i = 0; i < 3; i++) {
+    if (fromFormat[i] === 'YYYY' || fromFormat[i] === 'YY') {
+      year = dayOfYear[i];
+    }
+  }
+
+  if (year.length === 2 && year < 30) {
+    year = '20' + year;
+  }
+
+  if (year.length === 2 && year >= 30) {
+    year = '19' + year;
+  }
```

Comment:
This approach to converting the year does not account for the case where the year is already four digits long. It will incorrectly prepend '20' or '19' to a four-digit year. You should check the length of the year before performing these operations.